src/common/graphHelpers.js

```diff
@@ -56,35 +116,39 @@ const getLabel = ({ data, previousData, formattedDate, label, previousLabel }) =
  *  { x: 'Jun 1', y: 56, label: '56 Sockets on Jun 1 \r\n +5 from previous day' }
  *
  * @param data {Array}
+ * @param error {boolean}
  * @param startDate {string}
  * @param endDate {string}
  * @param label {string}
  * @param previousLabel {string}
- * @returns {Array}
+ * @returns {Object} Object array result converted { chartData: {...} chartDomain {...} }
  */
-const convertGraphUsageData = ({ data, startDate, endDate, label, previousLabel }) => {
+const convertGraphUsageData = ({ data, error = false, startDate, endDate, label, previousLabel }) => {
   let chartData = [];
+  let chartDomain = {};
 
+  if (error || !data || !data.length) {
```

Comment:
hmmm... i think my mind is trained to short circuit logic like this. Have always done null/empty checks at the top of methods so that logic following and future iterations of that logic following don't need to check. IIUC, you will be forcing an exception to happen if `data` is null/undefined (but maybe that can't happen in this case). If `data` is empty, then yes, for loop does not execute, but then you are checking null/empty elsewhere in your subsequent logic too. That being said, happy to refactor this bit if it's less readable for you or it does not make sense here... 

Overall, it seems like you are trying the combine the "error" API response scenario with the "data malformed" scenario...yes?

Reply:
kinda... the redux ensures that even in an "error" scenario we always receive, at minimum an empty array. from that perspective the need to do an "error" check kind of goes away